src/contexts/OfflineDataContext.jsx

```diff
@@ -0,0 +1,307 @@
+                    ? [...state.pendingChanges, { type: 'restore', data: action.payload }]
+                    : state.pendingChanges
+            };
+        case DATA_ACTIONS.CREATE_BACKUP:
+            return {
+                ...state,
+                lastBackupCreated: Date.now()
+            };
+        case DATA_ACTIONS.RESTORE_BACKUP:
+            return {
+                ...state,
+                posts: action.payload.data,
+                usingBackup: true,
+                backupTimestamp: action.payload.timestamp,
+                error: null
+            };
+        case DATA_ACTIONS.UPDATE_DELETED_POSTS_COUNT:
+            return {
+                ...state,
+                deletedPostsCount: action.payload
+            };
+        case DATA_ACTIONS.SET_OFFLINE_MODE:
+            return {
+                ...state,
+                isOffline: action.payload
+            };
+        case DATA_ACTIONS.SET_SYNC_STATUS:
+            return {
+                ...state,
+                syncStatus: action.payload
+            };
+        default:
+            return state;
+    }
+}
+
+// Provider component
+export function DataProvider({ children }) {
+    const [state, dispatch] = useReducer(dataReducer, initialState);
+
+    // Functie om posts te laden met fallback
+    const loadPosts = async () => {
+        dispatch({ type: DATA_ACTIONS.LOAD_POSTS_START });
+        
+        try {
+            // Probeer van server te laden
+            const data = await fetchPosts();
+            
+            if (data) {
+                dispatch({ type: DATA_ACTIONS.LOAD_POSTS_SUCCESS, payload: data });
+                dispatch({ type: DATA_ACTIONS.SET_OFFLINE_MODE, payload: false });
+                dispatch({ type: DATA_ACTIONS.SET_SYNC_STATUS, payload: 'idle' });
+                
+                // Maak initiële backup als die nog niet bestaat
+                if (!isBackupInitialized()) {
+                    saveInitialBackup(data);
+                }
+            }
+        } catch (error) {
+            console.error("Fout bij het ophalen van posts:", error);
+            dispatch({ 
+                type: DATA_ACTIONS.LOAD_POSTS_ERROR, 
+                payload: "Kon posts niet laden. Probeer de backup te herstellen." 
+            });
+            
+            // Ga naar offline mode
+            dispatch({ type: DATA_ACTIONS.SET_OFFLINE_MODE, payload: true });
+            dispatch({ type: DATA_ACTIONS.SET_SYNC_STATUS, payload: 'failed' });
+            
+            // Probeer backup te herstellen
+            const backup = getLatestBackup();
+            if (backup && backup.data) {
+                dispatch({ type: DATA_ACTIONS.SET_USING_BACKUP, payload: backup });
+            }
+        }
+    };
+
+    // Effect om posts te laden bij eerste render
+    useEffect(() => {
+        const loadPostsData = async () => {
+            await loadPosts();
+        };
+
+        loadPostsData();
+    }, []);
```

Comment:
Je hoeft hier niet nog een async functie met een await te creeëren puur om loadPosts aan te roepen. Alleen loadPosts(); binnen de useEffect zou genoeg zijn geweest. loadPosts is zelf al een async functie met een await. Deze await is niet nodig wanneer je loadPosts wilt aanroepen.